qt5-static Core component doesn’t work on its own

using `Widgets` instead of `Core` links successfully
```
project(qt5corelink)
cmake_minimum_required(VERSION 3.0)

find_package(Qt5 REQUIRED COMPONENTS Core)

add_executable(${PROJECT_NAME} main.cpp)
target_link_libraries(${PROJECT_NAME} Qt5::Core)
```
```
#include <QDebug>
#include <QtGlobal>

int main() {
    qDebug() << qVersion();
}
```
```
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x20a): undefined reference to `pcre2_match_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x27a): undefined reference to `pcre2_jit_stack_create_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x46d): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x56d): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5bc): undefined reference to `pcre2_pattern_info_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5cd): undefined reference to `pcre2_pattern_info_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5f2): undefined reference to `pcre2_pattern_info_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x60e): undefined reference to `pcre2_config_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x719): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x7da): undefined reference to `pcre2_compile_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x90e): undefined reference to `pcre2_jit_compile_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0xa22): undefined reference to `pcre2_substring_number_from_name_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0xc91): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0xd75): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0xee4): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x1063): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x114c): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x120d): undefined reference to `pcre2_compile_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x132c): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x13ed): undefined reference to `pcre2_compile_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x1567): undefined reference to `pcre2_get_error_message_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x16bc): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x177d): undefined reference to `pcre2_compile_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x189c): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x195d): undefined reference to `pcre2_compile_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x1a59): undefined reference to `pcre2_jit_compile_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x1b2a): undefined reference to `pcre2_jit_compile_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x1ee6): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x20b9): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x239e): undefined reference to `pcre2_substring_number_from_name_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x247e): undefined reference to `pcre2_substring_number_from_name_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x2567): undefined reference to `pcre2_substring_number_from_name_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x2637): undefined reference to `pcre2_substring_number_from_name_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x272b): undefined reference to `pcre2_substring_number_from_name_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x28fa): more undefined references to `pcre2_substring_number_from_name_16' follow
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x2c3e): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x2d3e): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x2ed5): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x2ffe): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x33ec): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x3509): more undefined references to `pcre2_code_free_16' follow
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x4929): undefined reference to `pcre2_match_context_create_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x4940): undefined reference to `pcre2_jit_stack_assign_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x494b): undefined reference to `pcre2_match_data_create_from_pattern_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x4a09): undefined reference to `pcre2_get_ovector_pointer_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x4a76): undefined reference to `pcre2_match_data_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x4a80): undefined reference to `pcre2_match_context_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x4ac2): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x4d80): undefined reference to `pcre2_pattern_info_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x4e39): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x4ef9): undefined reference to `pcre2_compile_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x506c): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5382): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5523): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x55e6): undefined reference to `pcre2_compile_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5797): undefined reference to `pcre2_match_context_create_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x57ac): undefined reference to `pcre2_jit_stack_assign_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x57b7): undefined reference to `pcre2_match_data_create_from_pattern_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5867): undefined reference to `pcre2_get_ovector_pointer_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x58d7): undefined reference to `pcre2_match_data_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x58df): undefined reference to `pcre2_match_context_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5912): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5b22): undefined reference to `pcre2_pattern_info_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5bdc): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5ca2): undefined reference to `pcre2_compile_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5e90): undefined reference to `pcre2_match_context_create_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5eac): undefined reference to `pcre2_jit_stack_assign_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5ebc): undefined reference to `pcre2_match_data_create_from_pattern_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5f7a): undefined reference to `pcre2_get_ovector_pointer_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5fe9): undefined reference to `pcre2_match_data_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x5ff3): undefined reference to `pcre2_match_context_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x6026): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x62a0): undefined reference to `pcre2_pattern_info_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x6477): undefined reference to `pcre2_match_context_create_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x648c): undefined reference to `pcre2_jit_stack_assign_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x6497): undefined reference to `pcre2_match_data_create_from_pattern_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x6562): undefined reference to `pcre2_get_ovector_pointer_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x65d8): undefined reference to `pcre2_match_data_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x65e0): undefined reference to `pcre2_match_context_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x6622): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x6880): undefined reference to `pcre2_pattern_info_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x6959): undefined reference to `pcre2_pattern_info_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x696f): undefined reference to `pcre2_pattern_info_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x6985): undefined reference to `pcre2_pattern_info_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x6fbb): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x722a): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x72cb): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x757a): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x76a5): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text+0x2e3): undefined reference to `pcre2_match_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text$_ZN14QThreadStorageIP20QPcreJitStackPointerE10deleteDataEPv[_ZN14QThreadStorageIP20QPcreJitStackPointerE10deleteDataEPv]+0x16): undefined reference to `pcre2_jit_stack_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text$_ZN28QExplicitlySharedDataPointerI25QRegularExpressionPrivateED1Ev[_ZN28QExplicitlySharedDataPointerI25QRegularExpressionPrivateED1Ev]+0x31): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text$_ZN18QSharedDataPointerI30QRegularExpressionMatchPrivateED1Ev[_ZN18QSharedDataPointerI30QRegularExpressionMatchPrivateED1Ev]+0xa6): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text$_ZN18QSharedDataPointerI38QRegularExpressionMatchIteratorPrivateE6detachEv[_ZN18QSharedDataPointerI38QRegularExpressionMatchIteratorPrivateE6detachEv]+0xc0): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text$_ZN18QSharedDataPointerI38QRegularExpressionMatchIteratorPrivateE6detachEv[_ZN18QSharedDataPointerI38QRegularExpressionMatchIteratorPrivateE6detachEv]+0x1d1): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text$_ZN18QSharedDataPointerI30QRegularExpressionMatchPrivateE13detach_helperEv[_ZN18QSharedDataPointerI30QRegularExpressionMatchPrivateE13detach_helperEv]+0x135): undefined reference to `pcre2_code_free_16'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qregularexpression.o):qregularexpression.cpp:(.text$_ZN18QSharedDataPointerI38QRegularExpressionMatchIteratorPrivateE13detach_helperEv[_ZN18QSharedDataPointerI38QRegularExpressionMatchIteratorPrivateE13detach_helperEv]+0xa5): more undefined references to `pcre2_code_free_16' follow
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qeventdispatcher_win.o):qeventdispatcher_win.cpp:(.text+0xc7d): undefined reference to `__imp_timeSetEvent'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qeventdispatcher_win.o):qeventdispatcher_win.cpp:(.text+0xd24): undefined reference to `__imp_timeKillEvent'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qeventdispatcher_win.o):qeventdispatcher_win.cpp:(.text+0xfa2): undefined reference to `__imp_timeSetEvent'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qeventdispatcher_win.o):qeventdispatcher_win.cpp:(.text+0x1b06): undefined reference to `__imp_timeKillEvent'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qeventdispatcher_win.o):qeventdispatcher_win.cpp:(.text+0x1df7): undefined reference to `__imp_timeKillEvent'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qeventdispatcher_win.o):qeventdispatcher_win.cpp:(.text+0x222e): undefined reference to `__imp_timeSetEvent'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qeventdispatcher_win.o):qeventdispatcher_win.cpp:(.text+0x2dd7): undefined reference to `__imp_timeKillEvent'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qcoreapplication_win.o):qcoreapplication_win.cpp:(.text+0x2e1): undefined reference to `GetFileVersionInfoSizeW'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qcoreapplication_win.o):qcoreapplication_win.cpp:(.text+0x32d): undefined reference to `GetFileVersionInfoW'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qcoreapplication_win.o):qcoreapplication_win.cpp:(.text+0x427): undefined reference to `VerQueryValueW'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qfilesystemengine_win.o):qfilesystemengine_win.cpp:(.text+0x3092): undefined reference to `__imp_GetUserProfileDirectoryW'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qfilesystemengine_win.o):qfilesystemengine_win.cpp:(.text+0x4089): undefined reference to `NetApiBufferFree'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qfilesystemengine_win.o):qfilesystemengine_win.cpp:(.text+0x40cb): undefined reference to `NetShareEnum'
D:/msys64/mingw64/qt5-static/lib/libQt5Core.a(qfilesystemengine_win.o):qfilesystemengine_win.cpp:(.text+0x419c): undefined reference to `NetApiBufferFree'
```

--- a/mingw-w64-qt5-static/0036-qt-5.3.2-win32-qt5-static-cmake-link-ws2_32-and--static.patch
+++ b/mingw-w64-qt5-static/0036-qt-5.3.2-win32-qt5-static-cmake-link-ws2_32-and--static.patch
@@ -10,6 +10,6 @@
 +    set(_isExe $<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>)
 +    # INTERFACE_LINK_LIBRARIES is used to pass a linker flag (-static)
 +    #                                      and a library (ws2_32)
-+    set_target_properties(Qt5::Core PROPERTIES \"INTERFACE_LINK_LIBRARIES\" \"$<${_isExe}:-static;ws2_32>\")
++    set_property(TARGET Qt5::Core APPEND PROPERTY INTERFACE_LINK_LIBRARIES \"$<${_isExe}:-static;ws2_32>\")
 +    unset(_isExe)
 +endif()

--- a/mingw-w64-qt5-static/0042-qt-5.4.0-static-cmake-also-link-plugins-and-plugin-deps.patch
+++ b/mingw-w64-qt5-static/0042-qt-5.4.0-static-cmake-also-link-plugins-and-plugin-deps.patch
@@ -1,16 +1,18 @@
 diff -urN qt-everywhere-opensource-src-5.6.0.orig/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in qt-everywhere-opensource-src-5.6.0/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in
 --- qt-everywhere-opensource-src-5.6.0.orig/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in	2015-08-26 15:31:26.194016900 +0100
 +++ qt-everywhere-opensource-src-5.6.0/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in	2015-08-26 15:21:30.905016900 +0100
-@@ -64,14 +64,17 @@
+@@ -58,14 +58,19 @@
      set(imported_location \"$${CMAKE_DLL_DIR}${LIB_LOCATION}\")
  !!ENDIF
      _qt5_$${CMAKE_MODULE_NAME}_check_file_exists(${imported_location})
 +    if (_Qt5$${CMAKE_MODULE_NAME}_LIB_DEPENDENCIES)
 +        set(_list_sep \";\")
 +    endif()
++    set_property(TARGET Qt5::Core APPEND PROPERTY
++        INTERFACE_LINK_LIBRARIES \"$<$<CONFIG:${Configuration}>:${_Qt5$${CMAKE_MODULE_NAME}_LIB_DEPENDENCIES}${_list_sep}${_Qt5$${CMAKE_MODULE_NAME}_STATIC_${Configuration}_LIB_DEPENDENCIES}>\"
++    )
      set_target_properties(Qt5::$${CMAKE_MODULE_NAME} PROPERTIES
 -        \"INTERFACE_LINK_LIBRARIES\" \"${_Qt5$${CMAKE_MODULE_NAME}_LIB_DEPENDENCIES}${_Qt5$${CMAKE_MODULE_NAME}_STATIC_${Configuration}_LIB_DEPENDENCIES}\"
-+        \"INTERFACE_LINK_LIBRARIES_${Configuration}\" \"${_Qt5$${CMAKE_MODULE_NAME}_LIB_DEPENDENCIES}${_list_sep}${_Qt5$${CMAKE_MODULE_NAME}_STATIC_${Configuration}_LIB_DEPENDENCIES}\"
          \"IMPORTED_LOCATION_${Configuration}\" ${imported_location}
  !!IF !isEmpty(CMAKE_LIB_SONAME)
          \"IMPORTED_SONAME_${Configuration}\" \"$${CMAKE_LIB_SONAME}\"

--- a/mingw-w64-qt5-static/PKGBUILD
+++ b/mingw-w64-qt5-static/PKGBUILD
@@ -101,7 +101,7 @@ _ver_base=5.11.1
 # use 5.6.1-1 hot fix : only the archive name was changed from *-5.6.1.tar.xz to *-5.6.1-1.tar.xz
 _hotfix=
 pkgver=${_ver_base//-/}
-pkgrel=2
+pkgrel=3
 arch=('any')
 pkgdesc="A cross-platform application and UI framework (mingw-w64${_namesuff})"
 url='https://www.qt.io/'
@@ -732,13 +732,13 @@ sha256sums=('39602cb08f9c96867910c375d783eed00fc4a244bffaa93b801225d17950fb2b'
             '7f5e092a5062db776a96ab32758ff175c3689b79902f5d9656f31029c33f66dd'
             '1ada266bc6cd033d770bc40ad879527db222563fc20cf98cf2848b84b2a0e503'
             '27e9d402126d313ff647066207038578c26a3cf0f60833ae5d76157d53712eae'
-            '29f13f7b44476bebfc5280a8481fa24926185860476ed0fd05f6b5a959b801d8'
+            '799af493e184e2e3a070e80bedf62417ff617de8f02e18b9fa20c7563dd9d93c'
             'c114d2df071b66e47f4bd43d335d48868fd5bcf9670f276b557edacd53fa92c0'
             'b9939efea7792a960cdc9404a515c6f0a1cd42ffbdb018d52f2b0206fffe17f5'
             '3ae9d64fb683036b3b1f91036378791e1ec4a10205c59e78761d22d8479c0540'
             'd720ac76035d56caea1f045dfb468a25865f9d4b22016cec5414f1c639a7697a'
             '914051840e01bf453e3db710d4471e6758e39e2cb87f8ffe3072fb90218c9f83'
-            'f5454bc38d57a734e5b36f6c54310576339628298d810628443f0a58a46e7477'
+            '5cebf00614f3bc101642ea3a2f8d14644ab75ea5168f63c2b0f5f090adc61f8a'
             '29ed4566d4c1853b70a5173c6391ed90f123c5121b5836f2d16f8478f6354f0a'
             '7d7a99ff45c6ae4f39be663d52bafc125970787f364e84593117e6a7d79f7eae'
             '72e46178f2f694d8408f22684d1a7d39394056cb574fc4855c1f38c9d51e7e6c'

--- a/mingw-w64-qt5/0036-qt-5.3.2-win32-qt5-static-cmake-link-ws2_32-and--static.patch
+++ b/mingw-w64-qt5/0036-qt-5.3.2-win32-qt5-static-cmake-link-ws2_32-and--static.patch
@@ -10,6 +10,6 @@
 +    set(_isExe $<STREQUAL:$<TARGET_PROPERTY:TYPE>,EXECUTABLE>)
 +    # INTERFACE_LINK_LIBRARIES is used to pass a linker flag (-static)
 +    #                                      and a library (ws2_32)
-+    set_target_properties(Qt5::Core PROPERTIES \"INTERFACE_LINK_LIBRARIES\" \"$<${_isExe}:-static;ws2_32>\")
++    set_property(TARGET Qt5::Core APPEND PROPERTY INTERFACE_LINK_LIBRARIES \"$<${_isExe}:-static;ws2_32>\")
 +    unset(_isExe)
 +endif()

--- a/mingw-w64-qt5/0042-qt-5.4.0-static-cmake-also-link-plugins-and-plugin-deps.patch
+++ b/mingw-w64-qt5/0042-qt-5.4.0-static-cmake-also-link-plugins-and-plugin-deps.patch
@@ -1,16 +1,18 @@
 diff -urN qt-everywhere-opensource-src-5.6.0.orig/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in qt-everywhere-opensource-src-5.6.0/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in
 --- qt-everywhere-opensource-src-5.6.0.orig/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in	2015-08-26 15:31:26.194016900 +0100
 +++ qt-everywhere-opensource-src-5.6.0/qtbase/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in	2015-08-26 15:21:30.905016900 +0100
-@@ -64,14 +64,17 @@
+@@ -58,14 +58,19 @@
      set(imported_location \"$${CMAKE_DLL_DIR}${LIB_LOCATION}\")
  !!ENDIF
      _qt5_$${CMAKE_MODULE_NAME}_check_file_exists(${imported_location})
 +    if (_Qt5$${CMAKE_MODULE_NAME}_LIB_DEPENDENCIES)
 +        set(_list_sep \";\")
 +    endif()
++    set_property(TARGET Qt5::Core APPEND PROPERTY
++        INTERFACE_LINK_LIBRARIES \"$<$<CONFIG:${Configuration}>:${_Qt5$${CMAKE_MODULE_NAME}_LIB_DEPENDENCIES}${_list_sep}${_Qt5$${CMAKE_MODULE_NAME}_STATIC_${Configuration}_LIB_DEPENDENCIES}>\"
++    )
      set_target_properties(Qt5::$${CMAKE_MODULE_NAME} PROPERTIES
 -        \"INTERFACE_LINK_LIBRARIES\" \"${_Qt5$${CMAKE_MODULE_NAME}_LIB_DEPENDENCIES}${_Qt5$${CMAKE_MODULE_NAME}_STATIC_${Configuration}_LIB_DEPENDENCIES}\"
-+        \"INTERFACE_LINK_LIBRARIES_${Configuration}\" \"${_Qt5$${CMAKE_MODULE_NAME}_LIB_DEPENDENCIES}${_list_sep}${_Qt5$${CMAKE_MODULE_NAME}_STATIC_${Configuration}_LIB_DEPENDENCIES}\"
          \"IMPORTED_LOCATION_${Configuration}\" ${imported_location}
  !!IF !isEmpty(CMAKE_LIB_SONAME)
          \"IMPORTED_SONAME_${Configuration}\" \"$${CMAKE_LIB_SONAME}\"

--- a/mingw-w64-qt5/PKGBUILD
+++ b/mingw-w64-qt5/PKGBUILD
@@ -101,7 +101,7 @@ _ver_base=5.11.1
 # use 5.6.1-1 hot fix : only the archive name was changed from *-5.6.1.tar.xz to *-5.6.1-1.tar.xz
 _hotfix=
 pkgver=${_ver_base//-/}
-pkgrel=2
+pkgrel=3
 arch=('any')
 pkgdesc="A cross-platform application and UI framework (mingw-w64${_namesuff})"
 url='https://www.qt.io/'
@@ -732,13 +732,13 @@ sha256sums=('39602cb08f9c96867910c375d783eed00fc4a244bffaa93b801225d17950fb2b'
             '7f5e092a5062db776a96ab32758ff175c3689b79902f5d9656f31029c33f66dd'
             '1ada266bc6cd033d770bc40ad879527db222563fc20cf98cf2848b84b2a0e503'
             '27e9d402126d313ff647066207038578c26a3cf0f60833ae5d76157d53712eae'
-            '29f13f7b44476bebfc5280a8481fa24926185860476ed0fd05f6b5a959b801d8'
+            '799af493e184e2e3a070e80bedf62417ff617de8f02e18b9fa20c7563dd9d93c'
             'c114d2df071b66e47f4bd43d335d48868fd5bcf9670f276b557edacd53fa92c0'
             'b9939efea7792a960cdc9404a515c6f0a1cd42ffbdb018d52f2b0206fffe17f5'
             '3ae9d64fb683036b3b1f91036378791e1ec4a10205c59e78761d22d8479c0540'
             'd720ac76035d56caea1f045dfb468a25865f9d4b22016cec5414f1c639a7697a'
             '914051840e01bf453e3db710d4471e6758e39e2cb87f8ffe3072fb90218c9f83'
-            'f5454bc38d57a734e5b36f6c54310576339628298d810628443f0a58a46e7477'
+            '5cebf00614f3bc101642ea3a2f8d14644ab75ea5168f63c2b0f5f090adc61f8a'
             '29ed4566d4c1853b70a5173c6391ed90f123c5121b5836f2d16f8478f6354f0a'
             '7d7a99ff45c6ae4f39be663d52bafc125970787f364e84593117e6a7d79f7eae'
             '72e46178f2f694d8408f22684d1a7d39394056cb574fc4855c1f38c9d51e7e6c'

